contracts/modules/TransferManager/GeneralTransferManager.sol

```diff
@@ -168,16 +150,21 @@ contract GeneralTransferManager is ITransferManager {
                 //Anyone on the whitelist can transfer, regardless of time
                 return (_onWhitelist(_to) && _onWhitelist(_from)) ? Result.VALID : Result.NA;
             }
+            if (_from == issuanceAddress && (whitelist[_to].canBuyFromSTO == 0) && _isSTOAttached()) {
+                return Result.NA;
+            }
             if (allowAllWhitelistIssuances && _from == issuanceAddress) {
```

Comment:
As a minor optimization, we can use nested if to test the 3 conditions that require `_from == issuanceAddress`

Reply:
Done